### ISSUE TITLE
remove-useless-arguments/destructuring false positive

### ISSUE THREAD

**Issue by EvgenyOrekhov** [coderaiser/putout#23]
```js
const foo = bar(({ baz }) => ({ baz }));
foo({ qux: "qux" });
```

Expected result: there should be no `remove-useless-arguments/destructuring` warning.

Actual result:

```text
2:7  error  "qux" is useless argument of a function "foo" (remove-useless-arguments/destructuring)  putout/putout
```

Looks like putout infers the signature of `foo` from the fat arrow function, but it should not do that. The signature of `foo` is unknown here.

**Comment by coderaiser** [comment]
Thank you :), fixed with 10ba4b4, landed in `@putout/plugin-remove-useless-variables v4.2.1` 🎉 . Is it works for you?

**Comment by EvgenyOrekhov** [comment]
Yes, it works correctly now, thanks!
